### PR TITLE
✅ Use Foundry cheatcodes instead of shims

### PR DESCRIPTION
**Pull request by pedrommaiaa**
## Description

Use `hevm.prank(user)` from Foundry instead of ERC\*User.sol to optimize gas usage in ERC\*.t.sol files. 

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge snapshot`?
- [x] Ran `npm run lint`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

**Comment by transmissions11** [comment]
amazing! been meaning to get around to this for a while

let's remove those User contracts while we're at it

**Comment by pedrommaiaa** [comment]
Looks good?

File: src/test/ERC20.t.sol [hunk 1]
```diff
@@ -417,7 +431,7 @@ contract ERC20Test is DSTestPlus {
         uint256 amount,
         uint256 deadline
     ) public {
-        deadline = bound(deadline, 0, block.timestamp - 1);
+        hevm.assume(deadline < block.timestamp - 1);
```
Comment (transmissions11) [inline src/test/ERC20.t.sol:434]: i still prefer bound to assume, can you revert all the places you used assume instead?

**Comment by transmissions11** [comment]
thank you so much @pedrommaiaa !!

**Pull request merged**